internal/juju/client.go

```diff
@@ -125,13 +126,18 @@ var isJAAS bool
 // IsJAAS uses a synchronisation object to only perform the check once and return the same result.
 func (sc *sharedClient) IsJAAS(defaultVal bool) bool {
 	checkJAASOnce.Do(func() {
+		isJAAS = defaultVal
 		conn, err := sc.GetConnection(nil)
 		if err != nil {
-			isJAAS = defaultVal
 			return
 		}
 		defer conn.Close()
-		isJAAS = conn.BestFacadeVersion("JIMM") != 0
+		jc := jaasApi.NewClient(conn)
+		_, err = jc.ListControllers()
+		if err == nil {
+			isJAAS = true
+			return
+		}
```

Comment:
question: why making this change?

Reply:
The `BestFacadeVersion` method finds the intersection between the set of facade versions received from the controller with a set of facade that the client supports. Asking for the "JIMM" facade always returns 0 because the Juju client is not aware of a JIMM facade. The client interface also doesn't expose a way of viewing the facades directly.

So the next best option is to make this JIMM specific call which does not require elevated permissions and won't fail unless JIMM is not working.